File: fq.game_service/src/main/java/com/fq/game_service/dao/Game.java
package com.fq.game_service.dao;

import jakarta.persistence.*;

import java.util.List;

@Entity
public class Game {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(nullable = false)
    private GameType gameType;

    @Column(nullable = false)
    private List<String> clueList;

    @ManyToOne(cascade = CascadeType.MERGE)
    private Jersey jersey;

    @ManyToOne(cascade = CascadeType.MERGE)
    private Player player;

    @ManyToOne(cascade = CascadeType.MERGE)
    private Composition composition;

/*
    Getters and Setters
*/

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public GameType getGameType() {
        return gameType;
    }

    public void setGameType(GameType gameType) {
        this.gameType = gameType;
    }

    public List<String> getClueList() {
        return clueList;
    }

    public void setClueList(List<String> clueList) {
        this.clueList = clueList;
    }

    public Jersey getJersey() {
        return jersey;
    }

    public void setJersey(Jersey jersey) {
        this.jersey = jersey;
    }

    public Player getPlayer() {
        return player;
    }

    public void setPlayer(Player player) {
        this.player = player;
    }

    public Composition getComposition() {
        return composition;
    }

    public void setComposition(Composition composition) {
        this.composition = composition;
    }
}

File: fq.back_office/src/app/shared/models/game/game.ts
import { Composition } from "../composition/composition";
import { GameType } from "../game-type/game-type";
import { Jersey } from "../jersey/jersey";
import { Player } from "../player/player";

export class Game {

    constructor(
        gameType: GameType,
        clueList: string[],
        jersey?: Jersey,
        player?: Player,
        composition?: Composition,
        id?: number,
    ) {
        this.id = id;
        this.gameType = gameType;
        this.clueList = clueList;
        this.jersey = jersey;
        this.player = player;
        this.composition = composition;
    }

    id?: number;
    gameType: GameType;
    clueList: string[];
    jersey?: Jersey;
    player?: Player;
    composition?: Composition;
}

